templates/nodered/nodered.service

```diff
@@ -20,8 +20,9 @@ EnvironmentFile=-/home/pi/.node-red/EnvironmentFile
 # uncomment next line if you need to wait for time sync before starting
 #ExecStartPre=/bin/bash -c '/bin/journalctl -b -u systemd-timesyncd | /bin/grep -q "systemd-timesyncd.* Synchronized to time server"'
 
-ExecStart=/usr/bin/env node-red-pi $NODE_OPTIONS $NODE_RED_OPTIONS
-#ExecStart=/usr/bin/env node $NODE_OPTIONS red.js $NODE_RED_OPTIONS
+Environment="NODE_RED_OPTIONS=-v --settings /etc/revpi/node-red-settings.js"
```

Comment:
we should also investigate, if we need to extend the User settings from `$HOME/.node-red/settings.js`

when the argument `--settings /etc/revpi/node-red-settings.js` is provided the user settings are not loaded at all.

But we can simply check if `$HOME/.node-red/settings.js` exists and merge the options

Reply:
I think we should stay with the TLS proxy solution for now. With this option any setup which depends on the plain http connection will still work. And we don't need to handle the customer config in any special way.

Reply:
I'm also not convinced, that the Node-RED server implements the encryption correctly. The Apache server on the other hand has been proven over a long time.

Reply:
for the TLS proxy we would also need to alter the config as we need to change the port node-red is running on.

This could be done with the `--define` argument which replaces only a single config value and not the whole settings file.
It still overwrites any value for the port the user might set in `$HOME/.node-red/settings.js`

Reply:
Overwriting the port config

```suggestion
Environment="NODE_RED_OPTIONS=-D uiPort=1881 -D uiHost=127.0.0.1"
```

Reply:
No, Node-RED will stay on 1880 for now. We will just add the proxy on port 41880.

Reply:
@LukasZech is the missing puzzle peace for https://gist.github.com/nbuchwitz/5c4d39b7ddd2c00afe8ce5eaacce47dd wich will allow to handle http and https with redirect without any interference of the nodered config.

Reply:
> I'm also not convinced, that the Node-RED server implements the encryption correctly. The Apache server on the other hand has been proven over a long time.

node-red does not implement any cryptography itself
 
it simply passes these values to the [native Node.js https server](https://nodejs.org/dist/latest-v18.x/docs/api/https.html#httpscreateserveroptions-requestlistener) which was release in **2011**
so it is quite battle-proven and well audited

Reply:
Best solution would be
1. reverse-proxy with SSL offload listening on public port 1880 
2. node-red without SSL listening to private/internal port 1881

My proposal was only meant as intermediate solution if the reverse-proxy is deemed to risky to be included in the bullseye release

Reply:
But having two software stacks doing the same job on one system does still make no sense to me.